netbox/dcim/tables/devices.py

```diff
@@ -145,7 +145,8 @@ class DeviceTable(TenancyColumnsMixin, ContactsColumnMixin, NetBoxTable):
         verbose_name=_('Name'),
         accessor=Accessor('label'),
```

Comment:
i think this might also need to be changed?
```suggestion
        accessor=Accessor('name'),
```

Reply:
Instead of switching the accessor to name, it should be removed entirely. It was added in bf836c9 to display a value other than `name`.

I believe removing this line and switching the template `DEVICE_LINK` from `value` to `record.label` might preserve displaying the label but fix sorting. Sorry I didn't catch this before.